ngraph/test/frontend/paddlepaddle/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME "paddlepaddle_tests")
+
+file(GLOB_RECURSE SRC ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
+
+add_executable(${TARGET_NAME} ${SRC})
+
+target_link_libraries(${TARGET_NAME} PRIVATE frontend_shared_test_classes)
+
+add_clang_format_target(${TARGET_NAME}_clang FOR_TARGETS ${TARGET_NAME})
+
+install(TARGETS ${TARGET_NAME}
+        RUNTIME DESTINATION tests
+        COMPONENT tests
+        EXCLUDE_FROM_ALL)
+
+# Test model generating
+ie_check_pip_package(paddlepaddle WARNING)
+
+set(TEST_PADDLE_MODELS_DIRNAME test_model_zoo/paddle_test_models)
+target_compile_definitions(${TARGET_NAME} PRIVATE -D TEST_PADDLE_MODELS_DIRNAME=\"${TEST_PADDLE_MODELS_DIRNAME}/\")
+
+# If 'paddlepaddle' is not found, code will still be compiled
+# but models will not be generated and tests will fail
+# This is done this way for 'code style' and check cases - cmake shall pass, but CI machine doesn't need to have
+# 'paddlepaddle' installed to check code style
+if (paddlepaddle_FOUND)
```

Comment:
May be it's better just to skip even tests compilation? Binary which cannot be run - useless.

E.g. if we have PDPD framework, tests are run. Otherwise, we can just print warning via `message(WARNING ..)`

Reply:
I've discussed this with @azhogov previously - there is a need to fail at some stage if user enables PDPD and tests, but there is no 'paddlepaddle' installed and models can't be generated.
1) Fail on 'cmake' stage is not good - even 'code style' job will fail as it does 'cmake'
2) Fail on 'build' stage is also not good - it we have some frontend enabled by default (e.g. onnx, tf) - user will have to install python packages for this frontend to make project buildable (even if user doesn't work on such frontends). Also it may be good to verify that code at least can be built on CI
3) Fail on 'unit tests run' stage - looks correct for me. On step 2 build will display warning that test models were not generated